python-package/lightgbm/sklearn.py

```diff
@@ -327,23 +342,38 @@ def fit(self, X, y,
             return (eval_name, eval_result, is_bigger_better)
                 or list of (eval_name, eval_result, is_bigger_better)
 
-            y_true: array_like of shape [n_samples]
-                The target values
-            y_pred: array_like of shape [n_samples] or shape[n_samples * n_class] (for multi-class)
-                The predicted values
-            weight: array_like of shape [n_samples]
-                The weight of samples
-            group: array_like
-                group/query data, used for ranking task
+            y_true: array-like of shape = [n_samples]
+                The target values.
+            y_pred: array-like of shape = [n_samples] or shape = [n_samples * n_classes] (for multi-class)
+                The predicted values.
+            weight: array-like of shape = [n_samples]
+                The weight of samples.
+            group: array-like
+                Group/query data, used for ranking task.
             eval_name: str
-                name of evaluation
+                The name of evaluation.
             eval_result: float
-                eval result
+                The eval result.
             is_bigger_better: bool
-                is eval result bigger better, e.g. AUC is bigger_better.
-        for multi-class task, the y_pred is group by class_id first, then group by row_id
-          if you want to get i-th row y_pred in j-th class, the access way is y_pred[j*num_data+i]
+                Is eval result bigger better, e.g. AUC is bigger_better.
+        For multi-class task, the y_pred is group by class_id first, then group by row_id.
+        If you want to get i-th row y_pred in j-th class, the access way is y_pred[j * num_data + i].
         """
+        if not hasattr(self, '_objective'):
+            self._objective = self.objective
+        if self._objective is None:
+            if isinstance(self, LGBMRegressor):
+                self._objective = "regression"
+            elif isinstance(self, LGBMClassifier):
+                self._objective = "binary"
+            elif isinstance(self, LGBMRanker):
+                self._objective = "lambdarank"
+            else:
+                raise ValueError("Unknown LGBMModel type.")
```

Comment:
I think it's better to warn, instead of raising an exception, because we can't make an instance of LGBMModel.

Reply:
I'm sorry that I was wrong.
Please ignore this point :p